actor/config.go

```diff
@@ -22,31 +22,47 @@ func defaultActorSystemConfig() Config {
 	}
 }
 
-func NewConfig() Config {
-	return defaultActorSystemConfig()
+type ConfigOption func(config Config) Config
```

Comment:
if it's func(*Config), then we don't need to return the value, avoiding copying values in and out
`type ConfigOption func(*Config)`